pgdog/src/backend/pool/shard.rs

```diff
@@ -194,19 +85,44 @@ impl ShardInner {
             }
         }
 
-        self.replicas.can_move_conns_to(&other.get_replicas())
+        self.replicas.can_move_conns_to(&other.replicas)
     }
 
-    fn duplicate(&self) -> Self {
+    /// Clone pools but keep them independent.
+    pub fn duplicate(&self) -> Self {
         Self {
-            primary: self.primary.as_ref().map(|primary| primary.duplicate()),
-            replicas: self.replicas.duplicate(),
-            rw_split: self.rw_split,
-            comms: ShardComms::default(),
+            inner: Arc::new(ShardInner {
+                primary: self
+                    .inner
+                    .primary
+                    .as_ref()
+                    .map(|primary| primary.duplicate()),
+                replicas: self.inner.replicas.duplicate(),
+                rw_split: self.inner.rw_split,
+                comms: ShardComms::default(), // Create new comms instead of duplicating
+            }),
         }
     }
 
-    async fn cancel(&self, id: &BackendKeyData) -> Result<(), super::super::Error> {
+    /// Bring every pool online.
+    pub fn launch(&self) {
+        ShardMonitor::run(self);
+        self.pools().iter().for_each(|pool| pool.launch());
```

Comment:
Since you're using pools inside the monitor, might want to launch them first. It's technically a race that will only happen on really fast hardware I believe.